### PR TITLE
adding gitlab cicd pipeline configuration

### PR DESCRIPTION
**Pull request by vsitnik**

### CODE REVIEW

File: cicd/gitlab/dot.gitlab-ci.yml
Comment (JAlvarezJarreta) [inline cicd/gitlab/dot.gitlab-ci.yml:47]: Based on the content, I assume this is just to check everything is configured properly?

File: cicd/gitlab/parts/python.gitlab-ci.yml
Comment (JAlvarezJarreta) [inline cicd/gitlab/parts/python.gitlab-ci.yml:64]: Since this covers the dependencies installed before (`python:prepare_env:venv:`), is there a need to have these 2 separated? I would expect all testing tasks (`pylint`, `mypy`, `pytest`, ...) to depend on this last job rather than the previous one.

File: cicd/gitlab/parts/python.gitlab-ci.yml
Comment (JAlvarezJarreta) [inline cicd/gitlab/parts/python.gitlab-ci.yml:69]: Prepping for the time when we can do integrations tests as well I see... I like it :nerd_face:

File: docs/cicd_gitlab.md
Comment (JAlvarezJarreta) [inline docs/cicd_gitlab.md:45]:
```suggestion
* We're trying to use `:` decided "namespace" qualifiers for stage and job names.
```

File: docs/cicd_gitlab.md
Comment (JAlvarezJarreta) [inline docs/cicd_gitlab.md:47]:
```suggestion
* Feel free to use template/generic jobs with names starting from `.` and `extends` keyword (see [`extends` description](https://docs.gitlab.com/ee/ci/yaml/#extends)
```

File: pyproject.toml
Comment (JAlvarezJarreta) [inline pyproject.toml:30]: Minimum version should say 3.8. I guess we forgot to apply this to this branch :see_no_evil:

**Review by JAlvarezJarreta — APPROVED** [review]
Three small typos, the rest looks great. Happy to see this moving forward.

**Review by JAlvarezJarreta — APPROVED** [review]
Three small typos, the rest looks great. Happy to see this moving forward.

File: cicd/gitlab/dot.gitlab-ci.yml
Comment (vsitnik) [reply cicd/gitlab/dot.gitlab-ci.yml:47]: it's more about checking env variable values

File: cicd/gitlab/parts/python.gitlab-ci.yml
Comment (vsitnik) [reply cicd/gitlab/parts/python.gitlab-ci.yml:64]: I'd rather prefer to have this as a separate task.  The previous one is more about tools and external bits installation

File: pyproject.toml
Comment (vsitnik) [reply pyproject.toml:30]: we forgot. let's wait until merge with an up-to-date

File: pyproject.toml
Comment (vsitnik) [reply pyproject.toml:111]: copied from @JAlvarezJarreta and @ens-LCampbell branch

File: cicd/gitlab/parts/python.gitlab-ci.yml
Comment (vsitnik) [reply cicd/gitlab/parts/python.gitlab-ci.yml:87]: It's a different job. With a different environment. I'd rather keep on checking based on my experience

File: pyproject.toml
Comment (vsitnik) [reply pyproject.toml:111]: just a seed configuration for pylint

**Review by ens-LCampbell — APPROVED** [review]

**Review by Dishalodha — APPROVED** [review]

**Pull request merged**
